### PR TITLE
Add QueryBuilders.matchNoneQuery(), #28679

### PR DESCRIPTION
**Pull request by m9aertner**
Small code change suggestion.
Someone with insights into ES documentation generation and QueryBuilders unit tests should review and make relevant further amendments. Thx.

### CODE REVIEW

**Comment by karmi** [comment]
Hi @m9aertner, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/elasticsearch/pull/28680.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by m9aertner** [comment]
Done.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/QueryDSLDocumentationTests.java [hunk 1]
```diff
@@ -249,6 +250,12 @@ public void testMatchAll() {
         // end::match_all
     }
 
+    public void testMatchNone() {
+        // tag::match_none
```
Comment (cbuescher) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/QueryDSLDocumentationTests.java]: These tests (and the tag itself) are really only useful in the context of the documentation where we extract example java snippets from the code. Take a look at https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/java-query-dsl-match-all-query.html and the corresponding documentation file https://github.com/elastic/elasticsearch/blob/6.2/docs/java-api/query-dsl/match-all-query.asciidoc to see what I mean.

In the query DSL docs at https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-match-all-query.html we group "match_all" and "match_none" on one page for convenience, I'd suggest doint that for the java api docs as well. Just adding another few lines to the match-all-query.asciidoc should be enough I think.

**Review by cbuescher — CHANGES_REQUESTED** [review]
@m9aertner thanks, looks good but I left a comment about also adding this example to the java api docs.

**Pull request closed without merging**